src-v5/carousel.tsx

```diff
@@ -15,476 +26,486 @@ import {
 } from './utils';
 import AnnounceSlide from './announce-slide';
 
-export const Carousel = (props: CarouselProps): React.ReactElement => {
-  const count = React.Children.count(props.children);
-  const [currentSlide, setCurrentSlide] = useState<number>(
-    props.autoplayReverse ? count - props.slidesToShow : props.slideIndex
-  );
-  const [animation, setAnimation] = useState<boolean>(false);
-  const [pause, setPause] = useState<boolean>(false);
-  const [dragging, setDragging] = useState<boolean>(false);
-  const [move, setMove] = useState<number>(0);
-  const [frameHeight, setFrameHeight] = useState<number>(0);
-  const visibleHeights = useRef<SlideHeight[]>([]);
-  const [keyboardMove, setKeyboardMove] = useState<KeyCodeFunction>(null);
-  const carouselWidth = useRef<number | null>(null);
-
-  const focus = useRef<boolean>(false);
-  const prevMove = useRef<number>(0);
-  const carouselEl = useRef<HTMLDivElement>(null);
-  const timer = useRef<ReturnType<typeof setTimeout> | null>(null);
-  const isMounted = useRef<boolean>(true);
-
-  useEffect(() => {
-    isMounted.current = true;
-    return () => {
-      isMounted.current = false;
+const Carousel = React.forwardRef<CarouselRef, CarouselProps>(
```

Comment:
I changed very few lines. Make sure to review this with "Hide whitespace" checked.

![image](https://user-images.githubusercontent.com/2937410/168988603-a6019ce3-faae-4722-b434-b74fb9526a21.png)